src/logic/documents/scorecards.js

```diff
@@ -53,55 +66,77 @@ export const downloadBlankScorecards = wcif => {
-          x2: pageWidth - scorecardMargin,
-          y2: pageHeight / 2,
-        }),
-        cutLine({
-          x1: pageWidth / 2,
-          y1: scorecardMargin,
-          x2: pageWidth / 2,
-          y2: pageHeight - scorecardMargin,
-        }),
-      ],
-    },
-  ],
-  pageMargins: [scorecardMargin, scorecardMargin],
-  content: {
-    layout: {
-      /* Outer margin is done using pageMargins, we use padding for the remaining inner margins. */
-      paddingLeft: i => (i % 2 === 0 ? 0 : scorecardMargin),
-      paddingRight: i => (i % 2 === 0 ? scorecardMargin : 0),
-      paddingTop: i => (i % 2 === 0 ? 0 : scorecardMargin),
-      paddingBottom: i => (i % 2 === 0 ? scorecardMargin : 0),
-      /* Get rid of borders. */
-      hLineWidth: () => 0,
-      vLineWidth: () => 0,
-    },
-    table: {
-      widths: ['*', '*'],
-      /* A4 page height minus vertical margins, divided into a half. */
-      heights: (pageHeight - 4 * scorecardMargin) / 2,
-      dontBreakRows: true,
-      body: chunk(scorecardList, 2),
+const scorecardsPdfDefinition = (scorecardList, imageData, paperSize) => {
+  const {
+    pageWidth,
+    pageHeight,
+    scorecardsPerRow,
+    scorecardsPerPage,
+  } = paperSizes[paperSize];
+  const imagePositions = [
+    /* Determined empirically to fit results table. */
+    { x: 60, y: 170 },
+    { x: 360, y: 170 },
+    { x: 60, y: 590 },
+    { x: 360, y: 590 },
+  ].slice(0, scorecardsPerPage);
+  const cutLines =
+    scorecardsPerPage === 4
+      ? {
+          canvas: [
+            cutLine({
+              x1: scorecardMargin,
+              y1: pageHeight / 2,
+              x2: pageWidth - scorecardMargin,
+              y2: pageHeight / 2,
+            }),
+            cutLine({
+              x1: pageWidth / 2,
+              y1: scorecardMargin,
+              x2: pageWidth / 2,
+              y2: pageHeight - scorecardMargin,
+            }),
+          ],
+        }
+      : {};
+
+  return {
+    background: [
+      ...(imageData === null
+        ? []
+        : imagePositions.map(absolutePosition => ({
+            absolutePosition,
+            image: imageData,
+          }))),
+      cutLines,
+    ],
+    pageSize: { width: pageWidth, height: pageHeight },
+    pageMargins: [scorecardMargin, scorecardMargin],
+    content: {
+      layout: {
+        /* Outer margin is done using pageMargins, we use padding for the remaining inner margins. */
+        paddingLeft: i => (i % scorecardsPerRow === 0 ? 0 : scorecardMargin),
+        paddingRight: i =>
+          i % scorecardsPerRow === scorecardsPerRow - 1 ? 0 : scorecardMargin,
+        paddingTop: i => (i % scorecardsPerRow === 0 ? 0 : scorecardMargin),
+        paddingBottom: i =>
+          i % scorecardsPerRow === scorecardsPerRow - 1 ? 0 : scorecardMargin,
+        /* Get rid of borders. */
+        hLineWidth: () => 0,
+        vLineWidth: () => 0,
+      },
+      table: {
+        widths: ['*'] * scorecardsPerRow,
```

Comment:
This actually returns `NaN` :D

```suggestion
        widths: Array(scorecardsPerRow).fill("*"),
```

Reply:
Ah, oops! Changed.